src/electionguard-ui/ElectionGuard.UI/ViewModels/CreateMultiTallyViewModel.cs

```diff
@@ -0,0 +1,353 @@
+                }
+            }
+        }
+    }
+
+    partial void OnSelectedKeyCeremonyChanged(KeyCeremonyRecord value)
+    {
+        // fill in the list of the elections that use the current key ceremony
+        _ = Task.Run(async () =>
+        {
+            var allElections = await _electionService.GetAllByKeyCeremonyIdAsync(value.KeyCeremonyId);
+            foreach (var item in allElections)
+            {
+                await LoadElectionData(item);
+            }
+            ElectionsLoaded = true;
+        });
+    }
+
+    private async Task LoadElectionData(Election election, string tallyId = "")
+    {
+        ElectionGuardException.ThrowIfNull(election.ElectionId, $"Election did not have an electionId {election.Id}");
+
+        var allUploads = await _ballotUploadService.GetByElectionIdAsync(election.ElectionId);
+        if (allUploads.Count() > 0)
+        {
+            var ballotCountTotal = 0L;
+            var ballotAddedTotal = 0L;
+            var ballotChallengedTotal = 0L;
+            var ballotSpoiledTotal = 0L;
+            var ballotDuplicateTotal = 0L;
+            var ballotRejectedTotal = 0L;
+
+            allUploads.ForEach((upload) =>
+            {
+                ballotCountTotal += upload.BallotCount;
+                ballotAddedTotal += upload.BallotImported;
+                ballotChallengedTotal += upload.BallotChallenged;
+                ballotSpoiledTotal += upload.BallotSpoiled;
+                ballotDuplicateTotal += upload.BallotDuplicated;
+                ballotRejectedTotal += upload.BallotRejected;
+            });
+            var electionItem = new ElectionItem
+            {
+                Election = election,
+                TallyId = tallyId,
+                BallotUploads = new(allUploads),
+                BallotAddedTotal = ballotAddedTotal,
+                BallotChallengedTotal = ballotChallengedTotal,
+                BallotDuplicateTotal = ballotDuplicateTotal,
+                BallotRejectedTotal = ballotDuplicateTotal,
+                BallotSpoiledTotal = ballotSpoiledTotal,
+                BallotCountTotal = ballotCountTotal
+            };
+            Elections.Add(electionItem);
+        }
+    }
+
+    private async Task FillKeyCeremonies()
+    {
+        var allKeys = await _keyCeremonyService.GetAllCompleteAsync();
+        foreach (var item in allKeys)
+        {
+            var count = await _electionService.CountByKeyCeremonyIdAsync(item.KeyCeremonyId);
+            if(count > 1)
+            {
+                KeyCeremonies.Add(item);
+            }
+        }
+    }
+
+    [RelayCommand]
+    private void SelectionChanged()
+    {
+        if(SelectedElections.Count != Elections.Count)
+        {
+            SelectAll = false;
+        }
+        CreateTalliesCommand.NotifyCanExecuteChanged();
+        JoinTalliesCommand.NotifyCanExecuteChanged();
+    }
+
+    private async Task<string> CreateNewTally(ElectionItem election, bool multi=false)

```

Comment:
naming ... `electionItem`

Reply:
Also this belongs on the `TallyRecord` type. All of this is static, based on params given.